lib/tlscontext.h

```diff
@@ -57,13 +57,21 @@ typedef enum
 typedef gint (*TLSSessionVerifyFunc)(gint ok, X509_STORE_CTX *ctx, gpointer user_data);
 typedef struct _TLSContext TLSContext;
 
+#define X509_MAX_CN_LEN 64
+#define X509_MAX_O_LEN 64
+#define X509_MAX_OU_LEN 32
+
 typedef struct _TLSSession
 {
   SSL *ssl;
   TLSContext *ctx;
   TLSSessionVerifyFunc verify_func;
   gpointer verify_data;
   GDestroyNotify verify_data_destroy;
+  int peer_info_found;
+  gchar peer_info_o[X509_MAX_O_LEN];
+  gchar peer_info_ou[X509_MAX_OU_LEN];
+  gchar peer_info_cn[X509_MAX_CN_LEN];
 } TLSSession;
```

Comment:
we may have a misunderstanding here, I didn't want you to inline these into the base struct, only to contain that. But this gave me an idea, what do you think of this:

```
 typedef struct _TLSSession
 {
   SSL *ssl;
   TLSContext *ctx;
   TLSSessionVerifyFunc verify_func;
   gpointer verify_data;
   GDestroyNotify verify_data_destroy;
   struct
   {
     gboolean info_found;
     gchar o[X509_MAX_O_LEN];
     gchar ou[X509_MAX_OU_LEN];
     gchar cn[X509_MAX_CN_LEN];
 } peer_info;
};
```

This way you have tls_session->peer_info.ou as the struct member. What do you think?